### ISSUE TITLE
upstream libc++ std::filesystem test fixes, check in test harness

### ISSUE THREAD

**Issue by DanAlbert** [android/ndk#1265]
I ended up with a bunch of patches that I need to upstream for testing https://github.com/android/ndk/issues/609. The implementation for the library is currently checked in, but the tests are waiting on that and a few other in-progress changes from upstream.

**Comment by DanAlbert** [comment]
Duplicate of #1530
